### DISCUSSION TITLE
about save checkpoint

### DISCUSSION THREAD

**Discussion by JinYuannn** [open-edge-platform/anomalib#2228] [Q&A] [answered]
During the training process, is the saved model checkpoint the model trained on the latest epoch or the model with the best evaluation? If it is the latter, what evaluation metric is it based on?

**Comment by abc-125** [comment, marked as answer]
I think the last epoch is saved, although you can use early stopping with any [available metrics](https://anomalib.readthedocs.io/en/latest/markdown/guides/reference/metrics/index.html) to get a model, which might perform better on a validation data.

**Comment by JinYuannn** [comment]
```python    
checkpoint_callback = ModelCheckpoint(
    filename='best-{epoch}-{pixel_F1Score:.3f}',
    monitor='pixel_F1Score',
    mode='max'
)

engine = Engine(
    max_epochs=400, 
    callbacks=[checkpoint_callback],
)
```
addition: You can use the ModelCheckpoint from the lightning library to save the checkpoint with the highest validation accuracy.

**Reply by samet-akcay** [reply]
Nice one @JinYuannn!
